### PR TITLE
Merge 24.0 into 24.1

### PR DESCRIPTION
**Pull request by mvdbeek**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mvdbeek** [comment]
I was :). Can you take a look at https://github.com/galaxyproject/galaxy/pull/18239/commits/0f9eab7869b43dc5ca289ec2209b828854210cc5 ? My thinking is that we need to separate the two error conditions, if submitting the value failed we shouldn't disable the rest of the interface

File: client/src/components/DatasetInformation/DatasetAttributes.vue [hunk 1]
```diff
@@ -26,7 +26,8 @@ const props = defineProps<Props>();
 const historyStore = useHistoryStore();
 
 const loading = ref(false);
-const messageText = ref("");
+const loadingFailed = ref(false);
+const messageText = ref<string | null>(null);
```
Comment (davelopez) [inline client/src/components/DatasetInformation/DatasetAttributes.vue]: > My thinking is that we need to separate the two error conditions, if submitting the value failed we shouldn't disable the rest of the interface

Yeah! That looks way better! Thank you!

Minor thing, undefined can serve here as well no need for extra null handling.
```suggestion
const messageText = ref<string>();
```

Comment (mvdbeek) [reply client/src/components/DatasetInformation/DatasetAttributes.vue]: don't we need to include undefined then in the type hint ?

Comment (davelopez) [reply client/src/components/DatasetInformation/DatasetAttributes.vue]: It should be inferred by leaving the value empty

image

Unless you want to use the more verbose type definition that imports `Ref` from vue:

```ts
const messageText: Ref<string | undefined> = ref();
```

**Pull request merged**